control/freqplot.py

```diff
@@ -117,65 +117,100 @@ def bode_plot(syslist, omega=None, dB=None, Hz=None, deg=None,
     # If argument was a singleton, turn it into a list
     if (not getattr(syslist, '__iter__', False)):
         syslist = (syslist,)
-
-    mags, phases, omegas = [], [], []
+    
+    if omega is None:
+        # Select a default range if none is provided
```

Comment:
Suggest using omega_num in this case, or raising an error if omega_nums is not None